NDice/DieLabeler.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NDice
+{
+	/// <summary>
+	/// Labels a Die roll result to a value of type TLabel
+	/// </summary>
+    public class DieLabeler<TLabel>
+    {
+		private readonly IDie _die = null;
+		private readonly IReadOnlyDictionary<int, TLabel> _labels = null;

```

Comment:
Default values aren't needed for any of these. Also, do we want to make labels public? Wouldn't it be useful to be able to access all labels?